### ISSUE TITLE
How to make Kolbot stop teleporting into mobs?

### ISSUE THREAD

**Issue by LithiumLovers** [kolton/d2bot-with-kolbot#124]
Hi. First of all, thank you all contributors for the amazing work!
My question: I've been running Kolbot without any problems. My sorc script is tweaked to fulfill my needs. Dodge is off. Dodge range is 15 (with dodge turned off). Sometimes my sorceress kill mobs with a safe distance, sometimes not. I will give you an example. When i'm doing Chaos Sanctuary with her, she keeps teleporting into melee range of Lord de Seis and Infector of Souls. How do i modify that? Because the bot is doing very well unless when he decides to teleport within close range. Is there any known fix to attack.js? Thank you very much.

**Comment by LithiumLovers** [comment]
I've tried turning off Config.TeleStomp = false; // Use merc to attack bosses if they're immune to attacks, but not to physical damage. Let's see what happens.

**Comment by LithiumLovers** [comment]
This issue has been fixed with this configuration. But even with telestomp set to false, my sorc keep on teleporting in the middle of Infector of Souls wherever he spawns. Do someone have a tweaked Diablo.js to share? I would be very thankful. Or maybe give me some tips because i have an average C  knowledge, which looks very similar to javascript. Ty.

**Comment by WiZzMaN47** [comment]
have u tried puttin changing dodge to true and make distance like 9 since sorc has tele?

**Comment by LithiumLovers** [comment]
Thanks for the reply. I have set dodge to false since my sorc doesn't stop teleporting and this messes up her cast. Main problem here seem to be Diablo.js. She teleports into De Seis and into Infector of Souls mob packs, making it a risk to run Diablo.js as sorc. She is trying to act as if she is Hammerdin. No matter if i turn dodge on, she keeps on suicide teleporting. Ty.

**Comment by WiZzMaN47** [comment]
i have no programmin skills but if u look at diablo.js ull c that 
De Seis has 2 map values for different layouts? 5325 & 5260
Vizier 7765
Inferctor 7825

this.vizlayout = this.getlayout(396, 5275);
this.seislayout = this.getlayout(394, 7773);
this.inflayout = this.getlayout(392, 7893);
 so just change the math equation looking for mosnters or just change the tele location

**Comment by LithiumLovers** [comment]
Alright, thank you Wizzman47 for the idea!

**Comment by dreadz** [comment]
Hey man i have the same issue with the sorc, how did u fix it ?

**Comment by DarkHorseDre** [comment]
Sorry for the necro - but this is still an issue for me - anyone know how to solve/has it been solved?  I've not seen reference to it on the SVN..

Cheers
DHD

**Comment by Kellen-Stuart** [comment]
What you could try is to set:
Dodge = true;
Config.DodgeRange = 20;  // further distance than default
Config.DodgeHP = 100; // meaning the sorc will dodge no matter what your health is

That said, the even better solution is to simply gear your sorc better so you don't have to worry about it. I find with dodge, my sorc isn't attacking as effectively since she's dodging the whole damn time.

If that doesn't work, let me know and I'll file a bug request.

**Comment by DarkHorseDre** [comment]
Hi @Kellen-Stuart, thanks for the response.  

I tried many permutations of the dodge settings, especially the range!  The problem is that the issue isn't the dodging, but more in the attack: the sorceress chooses to tele ontop of the mob instead of cast from a distance.  This is especially true of Lord de seis (hence my necro'ing this post as per OP).  It looks as if its a Hammerdin attack location, as the hammers would be cast just underneath the target.

if I could work out how to have the bot stand at a sensible distance to cast from when class = sorceress/caster (but not hammerdin), I think this would solve it.

As for the equipment - aside from the attack relying on a sorceress to tank (!), your suggestion doesn't account for new sorceresses with basic to intermediate equipment.  My sorc was getting pwned by zeis when tanking even with max res and block!

To reproduce: set any sorceress with tele skill to do CS and when it gets to de seis it usually jumps into the centre of the pack, whereas most players just kite.

**Comment by Kellen-Stuart** [comment]
Understood. 
I think teleport is linked to the Diablo script directly (not an overall sorceress setting like dodge).
Since you narrowed this down to Seis, that’s probably an attainable fix.
When I get a chance I’ll look into it
    	
    	

    	Get Outlook for iOS
    
  




On Wed, Aug 15, 2018 at 4:53 AM -0600, "DarkHorseDre" <notifications@github.com> wrote:












Hi @Kellen-Stuart, thanks for the response.


I tried many permutations of the dodge settings, especially the range!  The problem is that the issue isn't the dodging, but more in the attack: the sorceress chooses to tele ontop of the mob instead of cast from a distance.  This is especially true of Lord de seis (hence my necro'ing this post as per OP).  It looks as if its a Hammerdin attack location, as the hammers would be cast just underneath the target.


if I could work out how to have the bot stand at a sensible distance to cast from when class = sorceress/caster (but not hammerdin), I think this would solve it.


As for the equipment - aside from the attack relying on a sorceress to tank (!), your suggestion doesn't account for new sorceresses with basic to intermediate equipment.  My sorc was getting pwned by zeis when tanking even with max res and block!


To reproduce: set any sorceress with tele skill to do CS and when it gets to de seis it usually jumps into the centre of the pack, whereas most players just kite.

**Comment by DarkHorseDre** [comment]
Thanks man, sounds good!  I did end up looking at the diablo script but was too n00by to work out how to change the behaviour for that section.. I couldn't work out if attack.js took over or if the attack was solely within the diablo script.

Please note that I had not tried diablo with a sorc in maybe 2 months so please do test first (although I haven't seen any related changes, but I think there was some de seis optimisation?).

ps - I saw a note on another site where the project was requesting help with documenting the project. I have some doc experience and have some notes/tables I'd like to verify and share if needed.

**Comment by Kellen-Stuart** [comment]
Would you mind checking if the Diablo script still has that problem?
    	
    	

    	Get Outlook for iOS
    
  




On Wed, Aug 15, 2018 at 10:08 AM -0600, "DarkHorseDre" <notifications@github.com> wrote:












Thanks man, sounds good!  I did end up looking at the diablo script but was too n00by to work out how to change the behaviour for that section.. I couldn't work out if attack.js took over or if the attack was solely within the diablo script.


Please note that I had not tried diablo with a sorc in maybe 2 months so please do test first (although I haven't seen any related changes, but I think there was some de seis optimisation?).


ps - I saw a note on another site where the project was requesting help with documenting the project. I have some doc experience and have some notes/tables I'd like to verify and share if needed.

**Comment by DarkHorseDre** [comment]
Hi, sure - I have some key issues so may be a short while until I get time to fix.

Have the diablo seis/attack been changed recently?

**Comment by Kellen-Stuart** [comment]
I don’t know, but I’ve noticed my sorc teleporting into mobs in the Diablo script especially. I think the script was written for a Hammerdin. It might be tough to rewrite 
    	
    	

    	Get Outlook for iOS
    
  




On Sun, Aug 19, 2018 at 7:00 AM -0600, "DarkHorseDre" <notifications@github.com> wrote:












Hi, sure - I have some key issues so may be a short while until I get time to fix.


Have the diablo seis/attack been changed recently?

**Comment by DarkHorseDre** [comment]
Yeah that was my experience - not just de seis, but other mobs too.  I recently managed 2 runs - beat de seis twice but chickened to venom lord and infector soul in 2/2 runs.

Is there a dirty way to force kiting? like (psuedo):
if class = soso && skill == distance caster then position = few yards away?

I understand why it was written for hammerdin, but at the start of the season I couldn't afford one so used soso instead. had a hard time with soso but especially cs - dont run it anymore as a result!

**Comment by Kellen-Stuart** [comment]
I agree with you.

The issue is that i’ve only got experience making small modifications to this software. I know the code is sitting in Pather.js and Attack.js, however, how that code actually works is a total mystery to me.

**Comment by DarkHorseDre** [comment]
Ah ok, soz I assumed you were one of the devs.  Yeah I looked at all of that code and walked away lol

tbh not many people have reported this as they are probably using hammerdins!  I'm sure this will be picked up in time, but will likely be lesser priority..

**Comment by Kellen-Stuart** [comment]
I mean.. I’m professionally a dev. But not on this softwre

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by DarkHorseDre** [comment]
Yeah I thought you meant that - its harder to decipher other peoples code, especially when its not really documented or commented much!
I saw a few hints in the code but didnt fancy the high amount of testing and debugging of my changes...

**Comment by Kongolan** [comment]
It's not too hard to find the lines, you can easily just search by the config key notes like "Config.Dodge" and find the handle for it, it's not hard to find it, but the problem is how to check where to teleport. D2NT did this by checking the destination coordinates before actually teleporting to it, obviously because they had a proper maphack, i don't know how Kolbot works there though.

**Comment by DarkHorseDre** [comment]
Yep, and its not directly related to dodge - its the attack behaviour, which isn't just handled in attack.js but in the bot scripts too.  This is why Seis (with a soso especially) is a particular problem but many others aren't,  Travi is tough too as it will try and sit on their faces..

As for maphack, when I looked into this months ago, the bot did seem to check the for obstacles and mobs (maybe in attack.js?) but in other circumstances it seems to try to attack them from point blank range - it looks like a hammerdin attack position, which would explain a lot.

**Comment by Kellen-Stuart** [comment]
I figured out how to fix this.

Open the Diablo.js file and go to line 224 and replace the 4 line block of code with this block

if (this.seisLayout === 1) {
			Pather.moveTo(7771, 5200);
		} else {
			Pather.moveTo(7798, 5190);
		}

I changed the "Y" coordinate to be +4 and my sorc no longer teleports into De Seis. You can do the same thing with Vizier and the other Diablo mini bosses... Keep in mind, this is not a good spot for a Hammerdin, so maybe we should add a check if sorc or not?

**Comment by DarkHorseDre** [comment]
Nice work dude - I was gonna do this as per the earlier posts in this thread but realised a wider fix is needed as its not just de seis, and not just the diablo script (but seis failed often).  My fire-orb sorc chickened a lot and not just with de seis.. 

The class = soso check is defo worth doing as an cheap and easy control

**Comment by ninojuice** [comment]
As for Seis, switching the coordinate worked flawless. However, I can't find anywhere to change the script so that sorc doesn't tele into the middle of the mob still. Really teleports like a hammerdin. Any solutions yet?

**Comment by DarkHorseDre** [comment]
I think the answer may in be in attack.js, not looked.  I changed attacks for soso and it seems the range used for the attack may have something to do with it as I get it a lot less with certain skills.
which skills you using? orb?

**Comment by ninojuice** [comment]
> I think the answer may in be in attack.js, not looked. I changed attacks for soso and it seems the range used for the attack may have something to do with it as I get it a lot less with certain skills.
> which skills you using? orb?

Light sorc. wish I had my computer to scroll through attack.js :( I kind of tweaked dodge and distance to 10  which kind of helps her from staying in the middle of the mob too long but still...

**Comment by DarkHorseDre** [comment]
I think mine is alright now with light, blizz and orb after adding the changes above? its been a while.. 
soso is currently running nm dia with orb (previously blizz) and not landing in middle of mobs

**Comment by BigStupidNoob** [comment]
Howdy,

Was anyone able to find a permanent solution? I find that my Sorceress performs very well in all areas except for De Seis. Even though I adjusted the Y co-ordinate for Seis from...

```
		if (this.seisLayout === 1) {
			Pather.moveTo(7771, 5200);
		} else {
			Pather.moveTo(7798, 5186);
		}
```
                                **to**
```
		if (this.seisLayout === 1) {
			Pather.moveTo(7771, 5200);
		} else {
			Pather.moveTo(7798, 5190);
		}
```

I find that she may still sometimes get too close to the mobs or (depending on the map layout and spawn location of De Seis [whether the seal is oriented to the left or to the right]).

I think the safest solution will involve programming the bot to approach De Seis from afar and to attack from there as opposed to teleporting nearby then attacking.

Here is my character's dodge information:
```
	Config.Dodge = true; 
	Config.DodgeRange = 10; // Distance to keep from monsters.
	Config.DodgeHP = 100; // Dodge only if HP percent is less than or equal to Config.DodgeHP.
	Config.BossPriority = false; // Set to true to attack Unique/SuperUnique monsters first
	Config.ClearType = 0; // Monster spectype to kill in level clear scripts (ie. Mausoleum)
	Config.TeleStomp = false; // Use merc to attack bosses if they're immune to attacks
```
Great work so far and thanks for all the help. 👍

**Comment by DarkHorseDre** [comment]
IIRC it's attack.js (as referenced above) which decides the distance from target to attack.

**Comment by BigStupidNoob** [comment]
Does anyone have an idea as to which lines in the file Attack.js influence the distance a character will attack a target from?

If I were to guess, I would think it to be these lines:

```
	// Find an optimal attack position and move or walk to it
	getIntoPosition: function (unit, distance, coll, walk) {
		if (!unit || !unit.x || !unit.y) {
			return false;
		}

		if (walk === true) {
			walk = 1;
		}

		if (distance < 4 && (!unit.hasOwnProperty("mode") || (unit.mode !== 0 && unit.mode !== 12))) {
			//me.overhead("Short range");

			if (walk) {
				if (getDistance(me, unit) > 8 || checkCollision(me, unit, coll)) {
					Pather.walkTo(unit.x, unit.y, 3);
				}
			} else {
				Pather.moveTo(unit.x, unit.y, 0);
			}

			return !CollMap.checkColl(me, unit, coll);
		}
```

**Comment by Drakalkaisa** [comment]
I had success by just deleting those lines entirely, and letting the attack script do its job alone.  I think they were added as "ideal" starting positions for hammerdins, so the idea was to get him to move there, launch some attacks, then revert back to the attack script.

Im still dying alot but thats because my character is very weak, ill test it on nightmare to make sure it doesnt ruin the pathing and let you know how it goes, but I think it should be fine, since the script has a "check for stragglers" part later on that should send you back to kill anything missed.  

If you've edited those lines and STILL have issues make sure telestomp is also off and your not just teleporting in because of his immunities.  If this solution doesn't work im as hopeless as the rest of us  ¯\_(ツ)_/¯

**Comment by Drakalkaisa** [comment]
This is how my seal section looks now, I just straight up got rid of the moveto's

```
this.vizierSeal = function () {
		print("Viz layout " + this.vizLayout);
		this.followPath(this.vizLayout === 1 ? this.starToVizA : this.starToVizB);

		if (!this.openSeal(395) || !this.openSeal(396)) {
			throw new Error("Failed to open Vizier seals.");
		}

		if (!this.getBoss(getLocaleString(2851))) {
			throw new Error("Failed to kill Vizier");
		}

		return true;
	};

	this.seisSeal = function () {
		print("Seis layout " + this.seisLayout);
		this.followPath(this.seisLayout === 1 ? this.starToSeisA : this.starToSeisB);

		if (!this.openSeal(394)) {
			throw new Error("Failed to open de Seis seal.");
		}

		if (!this.getBoss(getLocaleString(2852))) {
			throw new Error("Failed to kill de Seis");
		}

		return true;
	};

	this.infectorSeal = function () {
		print("Inf layout " + this.infLayout);
		this.followPath(this.infLayout === 1 ? this.starToInfA : this.starToInfB);

		if (!this.openSeal(392)) {
			throw new Error("Failed to open Infector seals.");
		}


		if (!this.getBoss(getLocaleString(2853))) {
			throw new Error("Failed to kill Infector");
		}

		if (!this.openSeal(393)) {
			throw new Error("Failed to open Infector seals.");
		}

		return true;
	};
```

**Comment by Drakalkaisa** [comment]
I ended up having to also delete the throw error for "failed to kill" -it started working until I got to infector, now the sorc just jumped into that mob, so i tried a few things out and still had problems 50% of the time, reloaded the original file, same thing 50% of the time having some kind of "jump in the mob" issue.   I think its just the nature of the attack script, but in the packed corners of an open seal chaos sanctuary it ends up bad with their high damage.

**Comment by DarkHorseDre** [comment]
yeah you remind me that I looked into this a year or so ago and concluded the same thing - it behaves like a paladin.  someone else vouched and there were some workarounds.  the issue will still be here. y'all should search past issues..

**Comment by Drakalkaisa** [comment]
It's definitely the attack files not just the dibbles file, the "solution" seems to have just been a fluke of luck, judging from how my sorc acts in hell and the constant teleporting (instead of spam casting like a normal sorc) and even the patter of teleporting - yeah all indicates this bot was designed for a hammerdin.

**Comment by thisguy1986** [comment]
Gonna revive this ...

The best solution I have found is in the Pather.js and really the ultimate solution. If you have a hammerdin on bot you may , I don't know exactly how , probably somewhere in the config have them read a copy of this file patherH.js or whatever you want to rename it .

Anywho, go to your pather.js and around line 504 you will see movetounit ,var use teleport , under that is if (offx === undefined) line, sorry doing this on my phone, offx = 0 , and offy = 0 , change those variables to something like 7.
Now you will stop teleporting into mobs and however far away you set those variables .

**Comment by Qenozis** [comment]
Hello,
To better summarize the solution - few lines of the **Pather.js** have to be changed, so that before sorceress (or other non paladin class) is teleporting on top of the unit, first check if it is a monster and then add some offset values to X and Y coordinates, and if it's not a monster (like for example item to pick) leave it at zeros.

Sample code to change at "**moveToUnit**" function:
Line 505 (offX = 0;) to:

```
if (me.classid != 3) //not paladin
			{
				if (unit.type === 1) //is monster
				{
					offX = 6; //prevent sorc & similar teleporting on top of the target
				}
				else
				{
					offX = 0;
				}
			}
			else
			{
				offX = 0;
			}
```

Old line 509 (offY = 0;) to:

```
if (me.classid != 3) // not paladin
			{
				if (unit.type === 1) //is monster
				{
					offY = 6; //prevent sorc & similar teleporting on top of the target
				}
				else
				{
					offY = 0;
				}
			}
			else
			{
				offY = 0;
			}
```

The same changes can be made in function "**moveToPreset**".

With these changes, any non paladin classes will attack a monster from safe distance (if offX and offY were not provided) instead of jumping into mele range of possibly hard monster which can kill the char with heavy blows, like Seal Bosses in Chaos Sanctuary.

PS: for lazy ones or those, who like to see whole function after changes ;):

```
	moveToUnit: function (unit, offX, offY, clearPath, pop) {
		var useTeleport = this.useTeleport();

		if (offX === undefined) {
			if (me.classid != 3)// not paladin
			{
				if (unit.type === 1) //is monster
				{
					offX = 6; //prevent sorc & similar teleporting on top of the target
				}
				else
				{
					offX = 0;
				}
			}
			else
			{
				offX = 0;
			}
		}

		if (offY === undefined) {
			if (me.classid != 3)
			{
				if (unit.type === 1)
				{
					offY = 6; //prevent sorc & similar teleporting on top of the target
				}
				else
				{
					offY = 0;
				}
			}
			else
			{
				offY = 0;
			}
		}

		if (clearPath === undefined) {
			clearPath = false;
		}

		if (pop === undefined) {
			pop = false;
		}

		if (!unit || !unit.hasOwnProperty("x") || !unit.hasOwnProperty("y")) {
			throw new Error("moveToUnit: Invalid unit.");
		}

		if (unit instanceof PresetUnit) {
			return this.moveTo(unit.roomx * 5 + unit.x + offX, unit.roomy * 5 + unit.y + offY, 3, clearPath);
		}

		if (!useTeleport) {
			// The unit will most likely be moving so call the first walk with 'pop' parameter
			this.moveTo(unit.x + offX, unit.y + offY, 0, clearPath, true);
		}

		return this.moveTo(unit.x + offX, unit.y + offY, useTeleport && unit.type && unit.type === 1 ? 3 : 0, clearPath, pop);
	},
```

**Comment by Hyperflexion** [comment]
Switching the Pather.js lines to the lines above has worked ok so far. The bot does take longer to recognize NPCs while making room for items now. Is there anything I should adjust? Changing the Offx and Offy to 5 instead of 6 has helped a bit, but its still acting a bit weird.

**Comment by reg202** [comment]
Helpful info in this topic so far. I've had the same problem while running 3 characters together. The lead sorc does fine since she is opening the seals and is far enough when the monsters spawn. The barb follower is able to tank just fine. However the follower sorc sits right where the monsters spawn and sometimes chickens in the infector mob. I haven't attempted to check the code yet but it seems ideal to just fix the coordinates for that one spot (maybe seis too, just to be safe) instead of some of the more recent suggestions in the thread. I'll post again if I find anything else helpful

edit: I only watched a couple runs but it seems like infector layout 1 puts the followers in a worse spot than layout 2. I'll have to watch some more to confirm and then try to fix the coordinates

**Comment by reg202** [comment]
Again, I would not be changing settings and certainly not pather to fix a problem with one script. You're more likely to just screw up other things and the problem is with the Diablo.js script and nothing else. I just fixed my Infector coordinates and will explain how to do it for seis or viz if you need.

https://github.com/blizzhackers/documentation/blob/master/kolbot/Hotkeys.md/#hotkeys
The Home button on your keyboard brings up the console. Alt + Home lets you type things in. Typing getMouseCoords(1); will give you the coordinates of where you have the mouse pointed on screen. So check which seis or infector layout is giving you a problem (it will show which layout in the console) and then figure out where you want your character to be by using that command.

Here is the portion you want to change in the Diablo.js script for seis, for example. The first set of coordinates is for Layout 1, and the other is for layout 2. 

```
		if (this.seisLayout === 1) {
			Pather.moveTo(7771, 5196);
		} else {
			Pather.moveTo(7798, 5186);
```

For Infector 

```
		if (this.infLayout === 1) {
			delay(1);
		} else {
			Pather.moveTo(7928, 5295); // temp
```

For whatever reason, there is no coordinate for Layout 1, so I assume there is a previous instruction that tells the bot to go to the default spot, which is not a good spot. I changed mine to the following:

```
		if (this.infLayout === 1) {
			Pather.moveTo(7894, 5305);
		} else {
			Pather.moveTo(7928, 5295); // temp
```

This keeps the bot near the seal in layout 1 and stays safe

**Comment by SCTDM** [comment]
How my sorc used to die, but now hardly chickens in chaos clear. (edit: my sorc is a blizzard/ice blast + nova sorc with infinity)
First off I am assuming we are talking about "FastDiablo.js" here?

The standard code (in FastDiablo.js):

this.getBoss = function (name) {
		var i, boss,
			glow = getUnit(2, 131);

		for (i = 0; i < 24; i += 1) {
			boss = getUnit(1, name);

			if (boss) {
				this.chaosPreattack(name, 8);

				try {
					Attack.kill(name);
				} catch (e) {
					Attack.clear(10, 0, name);
				}

				Pickit.pickItems();

				return true;
			}

			delay(250);
		}

		return !!glow;
	};

My replacement code (inspired by Sealleader.js):

this.getBoss = function (name) {
		var i, boss,
			glow = getUnit(2, 131);

		for (i = 0; i < 16; i += 1) {
			boss = getUnit(1, name);

			if (boss) {
				this.chaosPreattack(name, 8);
				
				if (name === ("Lord De Seis")) {
				
				return Attack.clear(25, 0, name);
				
				} else {
				
				return Attack.clear(5, 0, name);
				
				}
			}

			delay(250);
		}

		return !!glow;
	};



So in short FastDiablo.js uses "Attack.kill" I use "Attack.clear". 
Change the first number in Attack.clear's parameters to adjust the clear range. A large range leads to a more cautious approach, a short range to a more aggressive.

Edit: the issues doesnt lie in the pathing coordinates it lies in how you approach the boss -> Attack.kill or Attack.clear

**Comment by famlove** [comment]
Just going to say how i setup my sorce and fixed most of these issues. 1. Was a tip previously stated about configuring pather to offset 6 in x and y when moving to unit(line 505). Next was setting dodge to true. putting range from monsters to 10 and setting hp to start dodge at 80. That way if the bot moves close to a unit while focusing on another it will dodge. Also set drink hp at same value of 80 that way a drinks a potion moves away and stops dodging to spam skills. Now if u want to fix were ur sorceres goes for seal bosses u can deleate or adjust the moveto in diablo config like previously stated or just delete it. Now something to add is the strings that the bot uses to clear called path cordinates.
	this.entranceToStar = [7794, 5517, 7791, 5491, 7768, 5459, 7775, 5424, 7817, 5458, 7777, 5408, 7769, 5379, 7777, 5357, 7809, 5359, 7805, 5330, 7780, 5317, 7791, 5293];
	this.starToVizA = [7759, 5295, 7734, 5295, 7716, 5295, 7718, 5276, 7697, 5292, 7678, 5293, 7665, 5276, 7662, 5314];
	this.starToVizB = [7759, 5295, 7734, 5295, 7716, 5295, 7701, 5315, 7666, 5313, 7653, 5284];
	this.starToSeisA = [7781, 5259, 7805, 5258, 7802, 5237, 7776, 5228, 7775, 5205, 7804, 5193, 7814, 5169, 7788, 5153];
	this.starToSeisB = [7781, 5259, 7805, 5258, 7802, 5237, 7776, 5228, 7811, 5218, 7807, 5194, 7779, 5193, 7774, 5160, 7803, 5154];
	this.starToInfA = [7809, 5268, 7834, 5306, 7852, 5280, 7852, 5310, 7869, 5294, 7895, 5295, 7919, 5290];
	this.starToInfB = [7809, 5268, 7834, 5306, 7852, 5280, 7852, 5310, 7869, 5294, 7895, 5274, 7927, 5275, 7932, 5297, 7923, 5313];
You can take these values and find the a consistent spot to back up to as needed and add to the end for followers. Or use them as a cord to jump to after opening seals to get a better engage on the seal bosses. Just my two cents glgl.
